api/main.py

```diff
@@ -101,6 +101,33 @@ async def get_node(node_id: str):
     return await db.find_by_id(Node, node_id)
 
 
+@app.get('/nodes')
+async def nodes():
+    return {'nodes': await db.find_all(Node)}
+
+
+@app.get('/node/{attr_name}/{attr_value}', response_model=Node)
+async def get_node(attr_name: str, attr_value: str):
+    return await db.find_by_attribute(Node, attr_name, attr_value)
+
+
+@app.get('/node/revision/{attr_name}/{attr_value}', response_model=Node)
+async def get_node(attr_name: str, attr_value: str):
+    return await db.find_by_attribute(Node, "revision", attr_name, attr_value)
+
+
+@app.get('/node', response_model=Node)
+async def get_node(request: Request):
+    params = str(request.query_params)
```

Comment:
The `Request.query_params` attribute is a dictionary with values already parsed.  So there's no need to convert it into a string and then split it up again.  It's summarised here: https://stackoverflow.com/a/67412008

So I think the URL handler could just be:
```python
return await db.find_by_attributes(Node, dict(request.query_params))
```